pkg/sql/opt/optbuilder/testdata/create_function

```diff
@@ -23,7 +26,33 @@ CREATE FUNCTION f(a workday) RETURNS INT LANGUAGE SQL AS $$ SELECT 1 $$
 ----
 create-function
  ├── CREATE FUNCTION f(IN a workday) RETURNS INT8 LANGUAGE SQL AS $$SELECT 1;$$
- └── no dependencies
+ └── dependencies
+      └── workday
+
+build
+CREATE FUNCTION f() RETURNS workday LANGUAGE SQL AS $$SELECT w FROM workdays$$
+----
+create-function
+ ├── CREATE FUNCTION f() RETURNS workday LANGUAGE SQL AS $$SELECT w FROM t.public.workdays;$$
+ └── dependencies
+      ├── workdays [columns: w]
+      └── workday
+
+build
+CREATE FUNCTION f() RETURNS STRING LANGUAGE SQL AS $$ SELECT 'MON'::workday::STRING $$
+----
+create-function
+ ├── CREATE FUNCTION f() RETURNS STRING LANGUAGE SQL AS $$SELECT 'MON'::STRING;$$
+ └── dependencies
+      └── workday
+
+build
+CREATE FUNCTION f() RETURNS STRING LANGUAGE SQL AS $$SELECT w::STRING FROM workdays$$
+----
+create-function
+ ├── CREATE FUNCTION f() RETURNS STRING LANGUAGE SQL AS $$SELECT w::STRING FROM t.public.workdays;$$
+ └── dependencies
+      └── workdays [columns: w]
```

Comment:
Similar question as the one I left in views: is it expected that the `workday` type is not a dependency here?